lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosValidation.cxx

```diff
@@ -0,0 +1,187 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Validation of a functional chaos expansion
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <algorithm>
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/FunctionalChaosValidation.hxx"
+#include "openturns/Os.hxx"
+#include "openturns/OSS.hxx"
+#include "openturns/DesignProxy.hxx"
+#include "openturns/LeastSquaresMethod.hxx"
+#include "openturns/KFoldSplitter.hxx"
+#include "openturns/MetaModelValidation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(FunctionalChaosValidation)
+
+static const Factory<FunctionalChaosValidation> Factory_FunctionalChaosValidation;
+
+/* Default constructor */
+FunctionalChaosValidation::FunctionalChaosValidation()
+  : MetaModelValidation()
+{
+  // Nothing to do
+}
+
+/* Default constructor */
+
+FunctionalChaosValidation::FunctionalChaosValidation(const FunctionalChaosResult & functionalChaosResult,
+                                                     const CrossValidationMethod cvMethod, 
+                                                     const UnsignedInteger & kParameter)
+  : MetaModelValidation(functionalChaosResult.getOutputSample()
+      , ComputeMetamodelCrossValidationPredictions(functionalChaosResult, cvMethod, kParameter))
```

Comment:
@jschueller : Is that implementation OK? I did that because there are no `MetaModelValidation.setOutputSample` and `MetaModelValidation.setMetamodelPredictions`. Hence, these parameters must be known at built time. If I implement the new `MetaModelValidation` methods, then I can compute the metamodel predictions in the body of the constructor and use the `MetaModelValidation.setMetamodelPredictions` method. This alternate implementation will have to modify `MetaModelValidation` and create a new private attribute `isMetamodelPredictionsInitialized_ ` initialized to false. When the `MetaModelValidation.setMetamodelPredictions` method is evaluated, the `isMetamodelPredictionsInitialized_` attribute is set to true. When the `draw` and `compute` methods are evaluated, the code must check if `isMetamodelPredictionsInitialized_` is true.